db/upgrade.php

```diff
@@ -304,5 +304,33 @@ function xmldb_tool_trigger_upgrade($oldversion) {
         upgrade_plugin_savepoint(true, 2021022300, 'tool', 'trigger');
     }
 
+    if ($oldversion < 2021030402) {
+
+        // Define field id to be added to tool_trigger_workflow_hist.
+        $table = new xmldb_table('tool_trigger_workflow_hist');
+        $field = new xmldb_field('userid', XMLDB_TYPE_INTEGER, '10', null, null, null, '0', 'event');
+
+        // Conditionally launch add field id.
+        if (!$dbman->field_exists($table, $field)) {
+            $dbman->add_field($table, $field);
+        }
+
+        // Update the new userid field with the associated user from the json eventdata.
+        $rs = $DB->get_recordset('tool_trigger_workflow_hist');
+        foreach ($rs as $record) {
```

Comment:
I dont think this is limited by the debug cleanup time, just the history time, which could be extremely heavy on some sites. Is it worth limiting it to the debug time to reduce the upgrade overhead? might need some live data to guide any more decisions here